website/files/models/github.py

```diff
@@ -14,3 +14,7 @@ class GithubFolder(GithubFileNode, Folder):
 
 class GithubFile(GithubFileNode, File):
     version_identifier = 'ref'
+
+    def touch(self, auth_header, revision=None, ref=None, branch=None, **kwargs):
+        revision = revision or ref or branch
```

Comment:
If more than one argument is specified (revision + branch), this will go with the first item that is specified- for example, if I ask for a past version on a particular branch, it will ask for the revision (but perhaps not the branch name).

The UI behaves well and as expected. Pinging @sloria to make sure this wouldn't cause any problems/unintended behavior if more than one revision selector were provided when this class were used internally. (as written, the arguments are ranked by descending priority. First one provided wins)